lib/console_web/controllers/event_controller.ex

```diff
@@ -68,5 +68,11 @@ defmodule ConsoleWeb.EventController do
     if event.device do
       Absinthe.Subscription.publish(ConsoleWeb.Endpoint, event, event_added: event.device.id)
     end
+    if event.gateway do
+      Absinthe.Subscription.publish(ConsoleWeb.Endpoint, event, event_added: event.gateway.id)
+    end
+    if event.channel do
+      Absinthe.Subscription.publish(ConsoleWeb.Endpoint, event, event_added: event.channel.id)
```

Comment:
do we need some more context here? we're just using id it looks like

Reply:
don't think so? the if statement should router the new event to the proper context. Even if the event belongs to more than 1 context, we should publish for each context as well right?

Reply:
the only difference i'm seeing between these 3 publishes is the `event_added: event.channel.id` vs `event_added: event.gateway.id` which is the id of the event parent.